### ISSUE TITLE
Issue about Reformer Encoder Decoder Architecture

### ISSUE THREAD

**Issue by guohanyang1994** [lucidrains/reformer-pytorch#122]
In Reformer Encoder Decoder Architecture, how are the parameters of the model updated, because I don’t seem to see the part about the optimizer and only `loss. backward()` exists in the code and the parameter of the learning rate in the code is missed.

**Comment by lucidrains** [comment]
@guohanyang1994 for a toy example of enc dec, you can largely substitute the instantiation here https://github.com/lucidrains/routing-transformer/blob/master/examples/toy_tasks/enc_dec_copy_task.py with ReformerEncDec and it should work

**Comment by lucidrains** [comment]
@guohanyang1994 I highly recommend Routing Transformer over Reformer, btw

**Comment by guohanyang1994** [comment]
> @guohanyang1994 I highly recommend Routing Transformer over Reformer, btw, even though the latter is more well known

Thanks for your advice. I tried this model, but the sequence length that the Routing Transformer can process seemed limited. I set the batch size as 16 and the sequence length as 1024, but it was out of GPU memory.
